Bumping react + fixing issue with timezone interface

sasankthapa opened the pull request:
Bumped react and react-dom version to latest 16.x.x. No clear instance of the whole app breaking. I still could save preferences, create groups and see meter data.

TimeZoneSelect is also edited so that it fixes the issue mentioned in PR #509 . 

Fixes #504.

Note merging this changes the node modules.

Commit:
Bumped react and react-dom to latest 16.x.x version and edited TimeZoneSelect to work on direct load of admin pages. Everything  else seemed to work fine, no drastic effect of upgrading react.

But there are more warnings that show up in the console regarding use of certain lifecycle method.

huss commented:
This addresses issue #504

huss reviewed (commented):
The changes look fine. Check and test show some issues but I believe they are all fixed when changes from earlier PRs are merged. I'm waiting to merge until those PRs are in development. At that time I will also do all checks/tests but expect the PR will readily be merged into development. Not changes are needed at this time.

huss committed:
Merge branch 'development' of https://github.com/OpenEnergyDashboard/OED into pr/sasankthapa/517

huss committed:
resolve linting issues

I am not sure why these did not show up earlier as they are not
part of the PR but from development merge.

huss reviewed (approved):
This has been reviewed and tested. It all looks good.
